src/contracts/yas_pool.cairo

```diff
@@ -43,7 +81,22 @@ mod YASPool {
 
     #[external(v0)]
     impl YASPoolImpl of IYASPool<ContractState> {
-        fn initialize(ref self: ContractState) { // TODO: implement
+        /// @notice Sets the initial price for the pool
+        /// @dev price is represented as a sqrt(amount_token_1/amount_token_0) Q64.96 value
+        /// @param sqrt_price_X96 the initial sqrt price of the pool as a Q64.96
+        fn initialize(ref self: ContractState, sqrt_price_X96: FixedType) {
+            assert(self.slot_0.read().sqrt_price_X96.is_zero(), 'AI');
+            let tick = TickMath::get_tick_at_sqrt_ratio(sqrt_price_X96);
+
+            self.slot_0.write(Slot0 { sqrt_price_X96, tick, fee_protocol: 0, unlocked: true });
+            self.emit(Initialize { sqrt_price_X96, tick });
```

Comment:
This way, the intention of updating the storage variable becomes clearer. What do you think of this suggestion?

```suggestion
            let mut slot_0 = self.slot_0.read(); 
            assert(slot_0.sqrt_price_X96.is_zero(), 'AI');

            slot_0.sqrt_price_X96 = sqrt_price_X96;
            slot_0.tick = TickMath::get_tick_at_sqrt_ratio(sqrt_price_X96);
            slot_0.fee_protocol = 0;
            slot_0.unlocked = true;
            self.slot_0.write(slot_0);
            
            self.emit(Initialize { sqrt_price_X96, tick: slot_0.tick });
```

Reply:
If we were using a parameter from Slot 0 of the storage, I would understand doing it this way, but I think using the write function is explicit enough.

From my point of view, doing it this way also seems to underutilize creating it directly with the Struct; it appears repetitive and doesn't contribute much.
```
slot_0.sqrt_price_X96 = sqrt_price_X96;
slot_0.tick = TickMath::get_tick_at_sqrt_ratio(sqrt_price_X96);
slot_0.fee_protocol = 0;
slot_0.unlocked = true;
self.slot_0.write(slot_0);
```

I like the idea of the first "if" statement; it's true that it's clearer that way.